pkg/docker/fast_push.go

```diff
@@ -128,12 +146,30 @@ func startUploadURL(objectType string, digest string) url.URL {
 	return uploadUrl
 }
 
-func uploadFile(objectType string, digest string, path string, token string) error {
+func verificationURL(objectType string, digest string, uuid string) url.URL {
+	verificationUrl := baseURL()
+	verificationUrl.Path = strings.Join([]string{"", "uploads", objectType, "sha256", digest, uuid, "verification"}, "/")
+	return verificationUrl
+}
+
+func checkVerificationStatus(req *http.Request, client *http.Client) (bool, error) {
+	checkResp, err := client.Do(req)
+	if err != nil {
+		return false, err
+	}
+	defer checkResp.Body.Close()
+	return checkResp.StatusCode == http.StatusOK, nil
+}
+
+func uploadFile(ctx context.Context, objectType string, digest string, path string, token string) error {
 	console.Debug("uploading file: " + path)
 
 	uploadUrl := startUploadURL(objectType, digest)
 	client := &http.Client{}
-	req, _ := http.NewRequest("POST", uploadUrl.String(), nil)
+	req, err := http.NewRequestWithContext(ctx, "POST", uploadUrl.String(), nil)
```

Comment:
nit: `http.MethodPost` instead of `"POST"`

Reply:
Resolved in [3975cbb](https://github.com/replicate/cog/pull/2128/commits/3975cbb80308e1dcd83f064040d47afad8111cc4)